js/modules/boost/boost-overrides.js

```diff
@@ -331,6 +331,21 @@ wrap(Series.prototype, 'processData', function (proceed) {
     }
 });
 
+
+// In heatmap, inactive state tries to modify opacity for other points.
+// Disable this for canvas (where opacity is disabled) to improve performance.
+if (seriesTypes.heatmap) {
+    wrap(
+        seriesTypes.heatmap.prototype.pointClass.prototype,
+        'setState',
+        function (proceed) {
+            if (!this.series.isSeriesBoosting) {
```

Comment:
This seems like it will disable all states when boosting, not just `inactive`. Is that on purpose?

Reply:
No, not really, can be added a condition for `inactive` state. 

Just to be on the same page, I realized that boosted heatmap does not have any states: https://www.highcharts.com/demo/heatmap-canvas (and simplified for select: https://jsfiddle.net/BlackLabel/Le6otns3/) so it's just a shortcut.

Reply:
> I realized that boosted heatmap does not have any states

If this can be confirmed by @TorsteinHonsi or another member that it is correct behavior, then I am fine with the current solution.